packages/liveblocks-core/src/types/Patchable.ts

```diff
@@ -0,0 +1,9 @@
+type OptionalKeys<T> = {
+  [K in keyof T]-?: {} extends Pick<T, K> ? K : never;
+}[keyof T];
+
+type MakeOptionalFieldsNullable<T> = {
+  [K in keyof T]: K extends OptionalKeys<T> ? T[K] | null : T[K];
+};
+
+export type Patchable<T> = Partial<MakeOptionalFieldsNullable<T>>;
```

Comment:
@marcbouchenoire I've replaced `PartialNullable<T>` with `Patchable<T>` across the board here. This new type is subtly closer to what we want when patching metadata.

Suppose you have:

```ts
type Meta = {
  color: "red" | "blue";
  resolved?: boolean;
}
```

Then:

```ts
PartialNullable<Meta>
// {
//   color?: "red" | "blue" | null;
//   resolved?: boolean | null;
// }
```

Whereas:

```ts
PartialNullable<Meta>
// {
//   color?: "red" | "blue";
//   resolved?: boolean | null;
// }
```

This ensures you cannot accidentally wipe mandatory fields when patching.